DSC19-78: Case Outcome

de-laz opened the pull request.

de-laz committed:
Add Case Outcome Questionnaire profile

Related Tickets:
 - https://jembiprojects.jira.com/browse/DSC19-59
 - https://jembiprojects.jira.com/browse/DSC19-78

de-laz committed:
Merge branch 'clinical-bundle' of github.com:openhie/covid-ig into dsc19-78-case-outcome

de-laz committed:
Add observation and Encounter resources

DSC19-78

de-laz committed:
Add questionnaire response example

DSC19-78

de-laz committed:
Add bundle example

DSC19-78

de-laz committed:
Update Hospitalization Encounter

The encounter was previously based on the base Encounter
profile. When applied, this commit will base the encounter
on the Case Report hospitalization encounter profile.

DSC19-78

de-laz committed:
Add Case Outcome encounter profile and example

DSC19-78

de-laz committed:
Add encounter resources to bundle

DSC19-78

de-laz committed:
Add Case Outcome Composition

DSC19-78

MattyJ007 commented on input/fsh/who-co-Observations.fsh:
```diff
@@ -0,0 +1,294 @@
+* class = http://terminology.hl7.org/CodeSystem/v3-ActCode#ACUTE
+* reasonCode = WhoCrCodeSystemReasonForTesting#CASE_CONTACT
+* period.start = "2021-05-12"
+* period.end = "2021-05-13"
```
Please move these Encounter resources into their own file.
Having each resource in its own file makes finding and maintaining the profiles a bit easier.

MattyJ007 commented on input/fsh/who-co-QuestionnaireResponse.fsh:
```diff
@@ -0,0 +1,74 @@
+
+* item[+].linkId = "outcome_contacts_followed_unknown"
+* item[=].text = "Number of contacts followed unknown"
+* item[=].answer[+].valueCoding = $cs-v2-0136#UNK
```
Missing newline at end of file

MattyJ007 commented on input/fsh/who-co-composition.fsh:
```diff
@@ -0,0 +1,58 @@
+* section[+].title = "exposure risk"
+* section[=].code = http://test.org/sectionCode#exposureRisk
+* section[=].entry[+] = Reference(WhoCoObservationOutcomeContactsFollowedUnknownExample)
+* section[=].entry[+] = Reference(WhoCoObservationOutcomeContactsFollowedExample)
```
Missing newline at end of file

MattyJ007 commented on input/fsh/who-co-Observations.fsh:
```diff
@@ -0,0 +1,294 @@
+* status = #final
+* valueCodeableConcept = WhoCoPositiveNegativeUnknown#positive
+
+Instance: WhoCoObservationOutcomeContactsFollowedExample
+InstanceOf: WhoCoObservationOutcomeContactsFollowed
+Usage: #example
+Title: "WHO CO Outcome Contacts Followed Example"
+Description: "Total number of contacts followed for this case"
+* status = #final
+* valueInteger = 5
```
@de-laz Could you please add an example where the integer value is -1 and include an `Observation.note: "unknown number of contacts follwed"`
This is just to show how we want `unknown` to be shown for this observation.

MattyJ007 commented on input/fsh/who-co-Observations.fsh:
```diff
@@ -0,0 +1,294 @@
+* status = #final
+* valueCodeableConcept = WhoCoHealthOutcome#death
+
+Instance: WhoCoObservationOutcomeLabDateExample
```
Please add this date value to the observation below as an `effectiveDateTime`

MattyJ007 commented on input/fsh/who-co-Observations.fsh:
```diff
@@ -0,0 +1,294 @@
+* status = #final
+* valueCodeableConcept = $cs-v2-0136#Y
+
+Instance: WhoCoObservationPatcourseAdmitDateExample
```
Please add this date value to the Observation above as an `effectiveDateTIme`.
Maybe you could also include an Observation in the negative (which would then exclude the date).
We won't necessarily use it in the `Bundle` but it'll be good to show the alternative form of the Observation.

MattyJ007 commented on input/fsh/who-co-Observations.fsh:
```diff
@@ -0,0 +1,294 @@
+* status = #final
+* valueCodeableConcept = $cs-v2-0136#Y
+
+Instance: WhoCoObservationOutcomeAsympDateExample
```
Please add this date value to the Observation above as an `effectiveDateTIme`.
Maybe you could also include an Observation in the negative (which would then exclude the date).
We won't necessarily use it in the Bundle but it'll be good to show the alternative form of the Observation.

MattyJ007 commented on input/fsh/who-co-Observations.fsh:
```diff
@@ -0,0 +1,294 @@
+* value[x] only integer
+* valueInteger 1..1
+
+Profile: WhoCoObservationOutcomeContactsFollowedUnknown
+Parent: Observation
+Id: who-co-observation-outcome-contacts-followed-unknown
+Title: "WHO CO Outcome Contacts Followed Unknown"
+Description: "WHO CO Outcome Contacts Followed Unknown Observation"
+* code = $LNC#96552-5
+* value[x] only CodeableConcept
+* valueCodeableConcept 0..1
+* valueCodeableConcept from WhoCrValueSetYesNoUnk
```
This observation can be deleted

MattyJ007 commented on input/fsh/who-co-Observations.fsh:
```diff
@@ -0,0 +1,294 @@
+Title: "WHO CO Outcome Lab Result"
+Description: "WHO CO Outcome Lab Result Observation"
+* code = $LNC#96552-5
+* value[x] only CodeableConcept
```
This Observation gets `effectiveDateTIme`

MattyJ007 commented on input/fsh/who-co-Observations.fsh:
```diff
@@ -0,0 +1,294 @@
+* value[x] only string
+* valueString 0..1
+
+Profile: WhoCoObservationOutcomeLabDate
```
This observation can be removed as its data will be included below

MattyJ007 commented on input/fsh/who-co-Observations.fsh:
```diff
@@ -0,0 +1,294 @@
+* valueCodeableConcept 1..1
+* valueCodeableConcept from WhoCrValueSetYesNoUnk
+
+Profile: WhoCoObservationPatcourseAdmitDate
```
This Observation can be removed as the data will be recorded above

MattyJ007 commented on input/fsh/who-co-Observations.fsh:
```diff
@@ -0,0 +1,294 @@
+Title: "WHO Case Outcome Patcourse Admit"
+Description: "WHO Case Outcome Patcourse Admit Observation"
+* code = $LNC#77974-4
+* value[x] only CodeableConcept
```
Add effectiveDateTIme

```suggestion
* value[x] only CodeableConcept
* effective[x] only DateTime
```

Not sure if my syntax is correct

MattyJ007 commented on input/fsh/who-co-Observations.fsh:
```diff
@@ -0,0 +1,294 @@
+* valueCodeableConcept 1..1
+* valueCodeableConcept from WhoCrValueSetYesNoUnk
+
+Profile: WhoCoObservationOutcomeAsympDate
```
This Observation can be removed - data is included above

MattyJ007 commented on input/fsh/who-co-Observations.fsh:
```diff
@@ -0,0 +1,294 @@
+Title: "WHO Case Outcome Outcome Asymp"
+Description: "WHO Case Outcome Outcome Asymp Observation"
+* code = $LNC#66421-9
+* value[x] only CodeableConcept
```
Add effectiveDateTime

MattyJ007 commented on input/fsh/who-co-Observations.fsh:
```diff
@@ -0,0 +1,294 @@
+Description: "WH Co Valueset Health Outcome"
+* codes from system WhoCoHealthOutcome
+
+Profile: WhoCoHospitalizationEncounter
+Parent: WhoCrEncounterHospitalAdmission
+Id: who-co-hospitalization-encounter
+Title: "WHO CO Hospitalization Encounter"
+Description: "WHO CO Hospitalization Encounter"
+* status = #finished
+* period 1..1
+* period.end 1..1
+
+Profile: WhoCoEncounter
+Parent: WhoCrEncounter
+Id: who-co-encounter
+Title: "WHO CO Encounter"
+Description: "WHO CO Encounter"
+* period 1..1
+* period.end 1..1
```
These should be in an Encounter file to make it easier to maintain

MattyJ007 commented on input/fsh/who-co-Observations.fsh:
```diff
@@ -0,0 +1,294 @@
+CodeSystem: WhoCoPositiveNegativeUnknown
+Title: "Positive/Negative/Unknown"
+Description: "Positive/Negative/Unknown"
+* #positive
+  "Positive"
+* #negative
+  "Negative"
+* #unknown
+  "Unknown"
+
+ValueSet: WhoCoValueSetPositiveNegativeUnknown
+Id: who-co-positive-negative-unknown
+Title: "Positive/Negative/Unknown"
+Description: "Positive/Negative/Unknown"
+* codes from system WhoCoPositiveNegativeUnknown
+
+CodeSystem: WhoCoHealthOutcome
+Title: "Health Outcome"
+Description: "Health Outcome"
+* #recovered_healthy
+  "Recovered/Healthy"
+* #not_recovered
+  "Not Recovered"
+* #death
+  "Death"
+* #unknown
+  "Unknown"
+* #other
+  "Other"
+
+ValueSet: WhoCoValueSetHealthOutcome
+Id: who-co-health-outcome
+Title: "Health Outcome"
+Description: "WH Co Valueset Health Outcome"
+* codes from system WhoCoHealthOutcome
```
Should these maybe go in the existing terminology file? Just to keep all related structure definitions together and to allow for reuse.
I'm not sure...
@rcrichton

de-laz replied:
@MattyJ007 do you think they should be in the existing file or a new file specific to `Case Outcome`?

MattyJ007 replied:
In the existing file should be fine - these options seem generic

MattyJ007 reviewed (changes requested):
@de-laz Looking good - just a few updates to the spec to cater to

de-laz committed:
Move CodeCystems and ValueSets to common file

DSC19-130

de-laz committed:
Move CO Encounters to their own file

DSC19-130

de-laz committed:
Add missing new lines to CO files

DSC19-130

de-laz committed:
Add Outcome Contacts Followed example

This commit will add an example Observation resource for an unknown
Outcome Contacts Followed value

DSC19-130

de-laz committed:
Move Asymp Date to Effective date

When applied, this commit will remove the Outcome Asymp Date
observation profile and add it as an effective date for Outcome Asymp
Observation.

DSC19-130

de-laz committed:
Remove Outcome contacts followed profile

This profile is no longer required according to the spec

DSC19-130

de-laz committed:
Move Outcome Lab Date to effective date

When applied, this commit will remove the Outcome Lab Date
Observation profile and add it as an effective date for Outcome Lab
result Observation profile.

DSC19-130

de-laz committed:
Move encounter examples to Encounters file

DSC19-130

de-laz committed:
Update observations file

This commit will remove Encounter examples from the
Observations fsh file

DSC19-130

de-laz committed:
Align Encounters with spec

DSC19-130

MattyJ007 commented on input/fsh/who-co-Observations.fsh:
```diff
@@ -111,6 +111,8 @@ Description: "WHO CO Outcome Contacts Followed Observation"
 * code = $LNC#96551-7
 * value[x] only integer
 * valueInteger 1..1
+* note 0..*
+* note.text 1..1
```
These definitions seem unnecessary as they are the same as the original spec
https://www.hl7.org/fhir/observation.html#resource

MattyJ007 commented on input/fsh/who-co-Observations.fsh:
```diff
@@ -0,0 +1,196 @@
+* valueCodeableConcept 0..1
+* valueCodeableConcept from WhoCoValueSetPositiveNegativeUnknown
+* effective[x] only dateTime
+* effectiveDateTime 1..1
```
```suggestion
* effectiveDateTime 0..1
```
In the spec this value won't always be supplied

MattyJ007 commented on input/fsh/who-co-Observations.fsh:
```diff
@@ -0,0 +1,196 @@
+* valueCodeableConcept = $cs-v2-0136#Y
+* effectiveDateTime = "2021-05-13"
+
+Instance: WhoCoObservationOutcomeAsympNoEffectiveDateExample
+InstanceOf: WhoCoObservationOutcomeAsymp
+Usage: #example
+Title: "WHO CO Outcome Asymp Example"
+Description: "Developed symptoms after time of specimen collection example without effective date"
+* status = #final
+* valueCodeableConcept = $cs-v2-0136#Y
```
In a more realistic case, I think the asymp date is only left out when the patient says "no".
But this is pedantic

```suggestion
Instance: WhoCoObservationOutcomeAsympNoEffectiveDateExample
InstanceOf: WhoCoObservationOutcomeAsymp
Usage: #example
Title: "WHO CO Outcome Asymp Example"
Description: "Did not develop symptoms after time of specimen collection example (no effective date)"
* status = #final
* valueCodeableConcept = $cs-v2-0136#N
```

MattyJ007 reviewed (changes requested):
Looking great 👍🏼 
Just some very minor things

de-laz committed:
Update input/fsh/who-co-Observations.fsh

Co-authored-by: Matt <matthew.dickie@jembi.org>

de-laz committed:
Update input/fsh/who-co-Observations.fsh

Co-authored-by: Matt <matthew.dickie@jembi.org>

MattyJ007 committed:
Merge two hospital admission Observations into one

The one observation was the date on which the admission happened. This
refers to the effective DateTime of the observation - not an independant
Observation.

DSC19-78

de-laz committed:
Remove note from Outcome Contacts Followed

This commit will remove the cardinality definition of note from
Outcome Contacts Followed observation profile. The base FHIR profile
already has a 1..1 cardinality defined

DSC19-130

MattyJ007 reviewed (approved):
👍🏼
